enhancement(search): simplify search docs and document opensearch backend (#1513)
opencloud-eu/opencloud: services/search/README.md

# Search

The search service is responsible for metadata and content extraction,
the retrieved date is indexed and made searchable.

The search service runs out of the box with the shipped default `basic` configuration.
No further configuration is needed.

## Search backends

To store and query the indexed data, a search backend is needed.

As of now, the search service supports the following backends:

- [bleve](https://github.com/blevesearch/bleve) (default)
- [opensearch](https://opensearch.org/)

### Bleve

Bleve is a lightweight, embedded full-text search engine written in Go and is the default search backend.

> It is easy to set up and requires no additional services to run.

Reviewer: Since we're listing the supported env vars for the opensearch backend, should we list the bleve one here as well (only `SEARCH_ENGINE_BLEVE_DATA_PATH` afaik)?

*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_HEADER=val`: HTTP headers to include in requests.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_CA_CERT=val` CA certificate for TLS connections.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_RETRY_ON_STATUS=val` HTTP status codes that trigger a retry.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_DISABLE_RETRY=val` Disable retries on errors.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_ENABLE_RETRY_ON_TIMEOUT=val`: Enable retries on timeout.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_MAX_RETRIES=val`: Maximum number of retries for requests.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_COMPRESS_REQUEST_BODY=val`: Compress request bodies.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_DISCOVER_NODES_ON_START=val`: Discover nodes on service start.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_DISCOVER_NODES_INTERVAL=val`: Interval for discovering nodes.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_ENABLE_METRICS=val`: Enable metrics collection.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_ENABLE_DEBUG_LOGGER=val`: Enable debug logging.
*   `SEARCH_ENGINE_OPEN_SEARCH_CLIENT_INSECURE=val`: Skip TLS certificate verification.

## Query language

> By default, [KQL](https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference) is used as a query language,

Reviewer: ```suggestion
By default, [KQL](https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference) is used as the query language.
```

> for an overview of how to write kql queries, please read the [microsoft documentation](https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference).

Reviewer: ```suggestion
For an overview of how to write kql queries, please read the [microsoft documentation](https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference).
```

*   NEAR operator
*   Date intervals

In [this ADR](https://github.com/owncloud/ocis/blob/docs/ocis/adr/0020-file-search-query-language.md) you can read why KQL was chosen.

## Content analysis / Extraction

> The search service supports the following content extraction services:

Reviewer: ```suggestion
The search service supports the following content extraction methods:
```
?

*   `Basic`: enabled by default, only provides metadata extraction.
*   `Tika`: needs to be installed and configured separately, provides content extraction for many file types.

> Note that files must be downloaded during the process of content extraction,

Reviewer: ```suggestion
Note that the file content has to be transferred to the search service internally for content extraction,
```
?

> Note that files must be downloaded during the process of content extraction,

Author: Note that the file content has to be transferred to the search service internally for content extraction,
which is resource-intensive and can lead to delays with larger documents.
